File: hackerrank.com/problem-solving/sock_merchant.scala
// https://www.hackerrank.com/challenges/sock-merchant/problem

import scala.io.StdIn._

object Solution {
  
  def sockMerchant(n: Int, ar: Array[Int]): Int = {
    return ar.distinct.map(i => ar.count(_ == i) / 2).sum
  }

  def sockMerchant1(n: Int, ar: Array[Int]): Int = {
    ar.groupBy(identity).mapValues(_.length / 2).values.sum
  }


  def main(args: Array[String]) {
    val n = readInt
    val ar = readLine.split(" ").map(_.trim.toInt)
    println(sockMerchant(n,ar))
  }
}

File: hackerrank.com/problem-solving/sock_merchant.py
# https://www.hackerrank.com/challenges/sock-merchant/problem

from collections import Counter

def sockMerchant(n, ar):
    c = Counter(ar)
    return sum([c[i] // 2 for i in c])
    
    
    
        
if __name__ == '__main__':
    n = int(input())
    ar = list(map(int, input().split()))
    print(sockMerchant(n, ar))

    